src/core/utils.cpp

```diff
@@ -427,4 +429,29 @@ generateModule(StringRef Routine, const Triple &Triple, StringRef Extension,
   }
 }
 
+IRChangesMonitor::IRChangesMonitor(const llvm::Module &M,
+                                   llvm::StringRef PassName)
+    : Mod(M) {
+  ObfuscationConfig *UserConfig = PyConfig::instance().getUserConfig();
+  if (UserConfig->has_report_diff_override()) {
+    this->UserConfig = UserConfig;
+    this->PassName = PassName.str();
+    llvm::raw_string_ostream(OriginalIR) << Mod;
+  }
+}
```

Comment:
```suggestion
IRChangesMonitor::IRChangesMonitor(const Module &M, StringRef PassName)
    : Mod(M), UserConfig(PyConfig::instance().getUserConfig()),
      PassName(PassName), ChangeReported(false) {
  if (UserConfig->has_report_diff_override())
    raw_string_ostream(OriginalIR) << Mod;
}
```
I'd personally go with this and check on `has_report_diff_override` for both, looks a bit cleaner using the initializer list.

Reply:
Address this in [be65525](https://github.com/open-obfuscator/o-mvll/pull/49/commits/be6552505736a62c5c9668b5ea6844add61d8c73) except that `PassName` string isn't initialized unconditionally. That would add an unnecessary heap alloc.

Reply:
Let's put `PassName` in the initializer list too, I'd prefer to favour readability. Most std::string implementations have some form of SSO, but regardless of this, the extra overhead due to the heap allocation is definitely not a concern in this context (and it's negligible too, considering we just have one invocation of `IRChangesMonitor`).